docs/apache-airflow/core-concepts/executor/index.rst

```diff
@@ -41,32 +49,175 @@ If you want to check which executor is currently set, you can use the ``airflow
+
+Some examples of valid multiple executor configuration:
+
+.. code-block:: ini
+
+    [core]
+    executor = 'LocalExecutor'
+
+.. code-block:: ini
+
+    [core]
+    executor = 'LocalExecutor,CeleryExecutor'
+
+.. code-block:: ini
+
+    [core]
+    executor = 'KubernetesExecutor,my.custom.module.ExecutorClass'
+
+
+.. note::
+    Using two instances of the _same_ executor class is not currently supported.
+
+To make it easier to specify executors on tasks and DAGs, executor configuration now supports aliases. You may then use this alias to refer to the executor in your DAGs (see below).
+
+.. code-block:: ini
+
+    [core]
+    executor = 'LocalExecutor,my.custom.module.ExecutorClass:ShortName'
+
+.. note::
+    If a DAG specifies a task to use an executor that is not configured, the DAG will fail to parse and a warning dialog will be shown in the Airflow UI. Please ensure that all executors you wish to use are specified in Airflow configuration on *any* host/container that is running an Airflow component (scheduler, workers, etc).
+
+Writing DAGs and Tasks
+^^^^^^^^^^^^^^^^^^^^^^
+
+To specify an executor for a task, make use of the executor parameter on Airflow Operators:
+
+.. code-block:: python
+
+    BashOperator(
+        task_id="hello_world",
+        executor="LocalExecutor",
+        bash_command="echo 'hello world!'",
+    )
+
+.. code-block:: python
+
+    @task(executor="LocalExecutor")
+    def hello_world():
+        print("hello world!")
+
+To specify an executor for an entire DAG, make use of the existing Airflow mechanism of default arguments. All tasks in the DAG will then use the specified executor (unless explicitly overridden by a specific task):
+
+.. code-block:: python
+
+    def hello_world():
+        print("hello world!")
+
+
+    def hello_world_again():
+        print("hello world again!")
+
+
+    with DAG(
+        dag_id="hello_worlds",
+        default_args={"executor": "LocalExecutor"},  # Applies to all tasks in the DAG
+    ) as dag:
+        # All tasks will use the executor from default args automatically
+        hw = hello_world()
+        hw_again = hello_world_again()
+
+.. note::
+    Tasks store the executor they were configured to run on in the Airflow database. Changes are reflected after each parsing of a DAG.
+
+Monitoring
+^^^^^^^^^^
+
+When using a single executor, Airflow metrics will behave as they were <2.9. But if multiple executors are configured then the executor metrics (``executor.open_slots``, ``executor.queued_slots``, and ``executor.running_tasks``) will be published for each executor configured, with the executor name appended to the metric name (e.g. ``executor.open_slots.<executor class name>``).
```

Comment:
We should also cross-reference it in https://github.com/apache/airflow/blob/main/docs/apache-airflow/administration-and-deployment/logging-monitoring/metrics.rst since that is the only place folks might look at for metrics

Reply:
We haven't really had a history of updating the metrics configuration page with any details like this. So the precedent feels a bit strange to set. But if you feel strongly then I can put a link in there!

Reply:
The more I thought on this, we need to add new rows to the metrics docs anyway, because the name for the new metrics will be different. They can be explained in the description column and possibly link to this section if we can't summarize them well in one or two sentences. @syedahsn can you add that to #40778

Reply:
>We haven't really had a history of updating the metrics configuration page with any details like this. So the precedent feels a bit strange to set. But if you feel strongly then I can put a link in there!

Hmm, what details specifically? Why strange? We define all the metrics on the metrics page, am I missing something?

>The more I thought on this, we need to add new rows to the metrics docs anyway, because the name for the new metrics will be different.

Yup, exactly.

>They can be explained in the description column and possibly link to this section if we can't summarize them well in one or two sentences.

Yup, we don't need to link it if you are already covering it in description :)

Reply:
> > We haven't really had a history of updating the metrics configuration page with any details like this. So the precedent feels a bit strange to set. But if you feel strongly then I can put a link in there!
> 
> Hmm, what details specifically? Why strange? We define all the metrics on the metrics page, am I missing something?

I meant that there is not metric description currently (or that I've ever seen) which links to a docs page. But either way the issue is resolved because the new rows added in Syed's PR describe the behaviour now! :)